packages/styled-components/src/models/StyledComponent.ts

```diff
@@ -37,6 +38,14 @@ import { DefaultTheme, ThemeContext } from './ThemeProvider';
 
 const identifiers: { [key: string]: number } = {};
 
+const isBrowser = typeof document !== 'undefined';
+// @ts-expect-error - this is fine
+const useInsertionEffect = React['useInsertion' + 'Effect']
```

Comment:
What's going on here?

Reply:
Emotion started using this pattern in https://github.com/emotion-js/emotion/pull/2651, in response to community reports that some bundlers got build errors when versions older than React 18 were used: https://github.com/emotion-js/emotion/issues/2649 🤔